spec/controllers/lectures_controller_spec.rb

```diff
@@ -78,7 +82,9 @@
 
     it "redirects to course overview for not joined students", :logged_lecturer do
       lecture = Lecture.create! valid_attributes_with_lecturer_with_course
-      login_student()
+      user = FactoryBot.create(:user, :student)
```

Comment:
sure you cant extract the user you're creating and logging in as them? would result in less duplicated lines

Reply:
No this is not possible, the `login_student` function creates a student if no student is given but does not return is